Nodes/puzzle_example/main.c

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (C) 2022 HAW Hamburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @{
+ *
+ * @file
+ * @brief       Example on using the puzzle_coap module
+ *
+ * @author      Bennet Blischke <bennet.blischke@haw-hamburg.de>
+ *
+ * @}
+ */
+
+#include <stdio.h>
+#include "shell.h"
+
+#include "puzzle_coap.h"
+
+#define MAIN_QUEUE_SIZE (4)
+static msg_t _main_msg_queue[MAIN_QUEUE_SIZE];
+
+/* forward declarations */
+static bool _solved(void);
+
+// Setting up the puzzle
+static puzzle_t puzzle = {
+    .puzzle_name = "Example puzzle",
+    .solved_handler = &_solved,
+};
+
+static bool _is_solved = false;
+
+
+int main(void)
+{
+    /* Initialise the puzzle coap module */
+    puzzle_init(&puzzle);
+
+    /* for the thread running the shell */
+    msg_init_queue(_main_msg_queue, MAIN_QUEUE_SIZE);
+    
+    /* start shell */
+    puts("All up, running the shell now");
+    char line_buf[SHELL_DEFAULT_BUFSIZE];
+    shell_run(NULL, line_buf, SHELL_DEFAULT_BUFSIZE);
+    return 0;
+}
+
+static int _cmd_solve(__attribute__((unused)) int argc, __attribute__((unused)) char **argv)
+{
+    _is_solved = !_is_solved;
```

Comment:
We usually deal with unused parameters like so
```suggestion
   (void) argc;
   (void) argv;
    _is_solved = !_is_solved;
```

Reply:
ACK [to be clear, I'm adopting the change]
ah, that's the reason behind the `(void) ctx` in the coap code, I should have realised that sooner.  I would prefer using the compiler attributes instead of tricking the compiler. Mhmm... I should probably run some test and look at the object files or maybe the binaries if it makes a difference. Most likely gcc/clang is smart enough to omit the code.

Okay, I googled a bit; Most programmers seem to prefer `(void)` over compiler attributes, claiming that it is more portable. However, gnu, clang and keil all perfectly understand the given attribute - suggesting that all common compilers can handle it. I did not find any information on how clang(or others) handles the `(void)`, so I either have to dig deeper into the documentation or do the suggested reverse analysis above. 🤔 Do you have further insight in this?  I would rather not waste a couple hours just to learn I could have asked! 😬